packages/pubsub/test/connection-pool.js

```diff
@@ -674,15 +750,162 @@ describe('ConnectionPool', function() {
     });
   });
 
+  describe('getAndEmitChannelState', function() {
+    var channelErrorEvent = 'channel.error';
+    var channelReadyEvent = 'channel.ready';
+    var channelReadyState = 2;
+    var fakeChannelState;
+    var dateNow;
+    var fakeTimestamp;
+
+    var fakeClient = {
+      getChannel: function() {
+        return {
+          getConnectivityState: function(shouldConnect) {
+            assert.strictEqual(shouldConnect, false);
```

Comment:
`asserts` should be kept within the `it`s, otherwise they're the first to go during someone's refactor. Could you keep this mock here, so that it still doesn't have to be written for each test, but do the assert in an explicit test?

Reply:
Fair enough!